modules/cats-effect/rules/src/main/scala/org/typelevel/fix/UnusedIO.scala

```diff
@@ -50,6 +51,8 @@ class UnusedIO extends SemanticRule("TypelevelUnusedIO") {
           checkSignature(outer, ref)
         case Term.ApplyInfix(_, op, _, _) =>
           checkSignature(outer, op)
+        case Term.Apply(fn @ Term.Name(_), _) if IOCompanionSym.matches(fn) =>
+          Patch.lint(UnusedIODiagnostic(outer))
         case Term.Apply(fn @ Term.Name(_), _) =>
           checkSignature(outer, fn)
```

Comment:
Why as a separate case, rather than folding it into the `checkSignature` call below?

Reply:
Unfortunately this is a bit of a special case - scalameta does not know that this gets desugared to a `def apply` method call, it just knows that the `IO` companion object is being applied to something.
So if we try to check the signature of `fn` here, it's just a `ClassSignature` of the companion object rather than a particular method of the companion object.
It seemed to make more sense to call this out as a special case here since the class signature isn't too useful.